exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -624,6 +666,72 @@ object SchemaUtils {
         return toCreate.toList()
     }
 
+    /**
+     * @param scriptName The name to be used for the generated migration script.
+     * @param scriptDirectory The directory in which to create the migration script.
+     * @param withLogs By default, a description for each intermediate step, as well as its execution time, is logged at
+     * the INFO level. This can be disabled by setting [withLogs] to `false`.
+     *
+     * @return The generated migration script.
+     *
+     * This function simply generates the migration script without applying the migration. The purpose of it is to show
+     * the user what the migration script will look like before applying the migration.
+     */
+    @ExperimentalDatabaseMigrationApi
+    fun generateMigrationScript(vararg tables: Table, scriptDirectory: String, scriptName: String, withLogs: Boolean = true): File {
```

Comment:
Some points:

1. I'm guessing this isn't the final form of the KDocs? The last 2 lines read like they're more for us than for the user (mostly the last sentence).
2. Maybe it's implied/obvious, but should the `scriptDirectory` emphasize that it's the absolute path that will be used?
3. This function will technically allow no table arguments to be passed. In this event, does an empty file still get generated? Or should an exception be thrown early?

Reply:
1. Should I simply omit "the user"? **"Its purpose is to show what the migration script will look like before applying the migration."**
2. It's not the absolute path, but the path from repository root. I will add that to the KDoc.
3. An empty file still gets generated. I like the idea of throwing an exception early to let the user know immediately. What exception do you think would be fitting here?

Reply:
1. Yes
2. Sounds good
3. Maybe an `IllegalArgumentException` from `require()`?